src/ownership.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+
+import cockpit from 'cockpit';
+
+import { FileInfo } from './fsinfo';
+
+interface Id {
+    id: number;
+    name: string | null;
+}
+
+interface IdInfo {
+    uid: Id;
+    gid: Id;
+    groups: Id[];
+}
+
+function parse_id(content: string): Id {
+    const match = content.match(/^(\d+)\(([^)]+)\)$/);
+    if (!match) {
+        throw new Error(`invalid id ${content}`);
+    }
+
+    const id = Number(match[1]);
+    const name = match[2] === "unbound" ? null : match[2];
+
+    return { id, name };
+}
+
+export function useId(): IdInfo | null {
+    /* XXX This should be part of the cockpit UserInfo */
```

Comment:
Starting to wonder if we really need the GID of all groups at all. In what scenario is the group name not known but the uid is?

Reply:
Expiremented and yeah it's very easy to get into this:
```
builder:x:1002:
:x:1003:admin
[root@fedora-40-127-0-0-2-2201 ~]# id admin
uid=1000(admin) gid=1000(admin) groups=1000(admin),10(wheel),100(users),1003()
```

Reply:
Ya, I don't think we actually use the "groups" information, in the end.  I added it for completeness.  This will all hopefully die when we expand the userinfo interface in Cockpit proper...

Reply:
Alright so I tested this with a weird /etc/group and it fails:
```
Failed to parse output of "id" command: uid=1000(admin) gid=1000(admin) groups=1000(admin),10(wheel),100(users),1003() context=unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023
: Error: invalid id 1003()
```